swap/src/protocol/alice/event_loop.rs

```diff
@@ -459,14 +465,21 @@ impl EventLoopHandle {
             .respond(())
             .context("Failed to acknowledge receipt of encrypted signature")?;
 
-        Ok(request.tx_redeem_encsig)
+        Ok((request.tx_redeem_encsig, request.swap_id))
     }
 
-    pub async fn send_transfer_proof(&mut self, msg: monero::TransferProof) -> Result<()> {
+    pub async fn send_transfer_proof(
+        &mut self,
+        swap_id: Uuid,
```

Comment:
Same goes here, I don't think the swap function (which is the user of this method) should have control over this swap-id. Instead, we should associate this at the level where we construct the channels such that whatever gets sent through this channel is automatically associated with the right swap-id on the receiver end.